src/main/java/teammates/ui/controller/AdminSearchPageAction.java

```diff
@@ -266,17 +266,15 @@ private String findAvailableInstructorGoogleIdForCourse(String courseId){
         if(instructorList == null || instructorList.isEmpty()){
             return googleId;
         }
-        
-        for(InstructorAttributes instructor : instructorList){
-          
-            if(instructor.googleId != null){
-               googleId = instructor.googleId;
-               if(instructor.isAllowedForPrivilege(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)){            
-                   break;
-               } 
-            }            
+
+        for (InstructorAttributes instructor : instructorList) {
+
+            if (instructor.isRegistered() && instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR)) {
```

Comment:
If there is no instructor with the same permissions as the Co-owner, how should we select a instructor for the link? An instructor with privilege to modify instructors can add all the permissions to himself, so `INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR` is essentially the only permission that a Co-owner needs.